src/tools/miri/tests/pass/async-closure-drop.rs

```diff
@@ -0,0 +1,40 @@
+#![feature(async_closure, noop_waker, async_fn_traits)]
```

Comment:
Moved the old test from `async-closure.rs` -> `async-closure-drop.rs` and then added a new test that tests the calling at `async-closure.rs`. I wish git were smarter about renames like that lol.